interceptor_test.go

```
@@ -0,0 +1,236 @@
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/corazawaf/coraza/v3"
+	"github.com/corazawaf/coraza/v3/types"
+	"github.com/stretchr/testify/require"
+)
+
+// fakeHijackConn is a minimal net.Conn returned by the fake hijacker.
+type fakeHijackConn struct{ net.Conn }
```

golangci-lint (2.12.2)

[error] type fakeHijackConn is unused

(unused)